pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -427,25 +403,123 @@ func (b *EndpointBuilder) BuildLocalityLbEndpointsFromShards(
 	if len(locs) >= 2 {
 		sort.Strings(locs)
 	}
-	for _, k := range locs {
-		locLbEps := localityEpMap[k]
+	for _, locality := range locs {
+		locLbEps := localityEpMap[locality]
 		var weight uint32
+		var overflowStatus bool
 		for _, ep := range locLbEps.llbEndpoints.LbEndpoints {
-			weight += ep.LoadBalancingWeight.GetValue()
+			weight, overflowStatus = addUint32(weight, ep.LoadBalancingWeight.GetValue())
 		}
 		locLbEps.llbEndpoints.LoadBalancingWeight = &wrapperspb.UInt32Value{
 			Value: weight,
 		}
+		if overflowStatus {
+			log.Warnf("Sum of localityLbEndpoints weight is overflow: service:%s, port: %d, locality:%s",
+				b.service.Hostname, b.port, locality)
+		}
 		locEps = append(locEps, locLbEps)
 	}
 
 	if len(locEps) == 0 {
 		b.push.AddMetric(model.ProxyStatusClusterNoInstances, b.clusterName, "", "")
 	}
 
+	// Apply the Split Horizon EDS filter, if applicable.
+	locEps = b.EndpointsByNetworkFilter(locEps)
```

Comment:
Suggest not apply to cds endpoint, just keep as before. Not quite sure whether we set the network label correct

Reply:
I'm not sure what you mean. The original motivation (for me) was to get multi-network working for non EDS. It's possible I have a ServiceEntry that has mixed IPs/Hostnames so it must be a DNS cluster. I want the IPs to get mapped.

Reply:
I mean for cds, we donot apply filter now. This pr is already big, if we want to apply network filter for cds endpoint, we can do it later